GoogleForCreators/web-stories-wp
# Dashboard: display template details in modal

## Context


Explore templates -> See Details looks like a modal, but isn't. Let's update it to be an actual modal, complete with navigation that makes sense for a modal. ie. `esc` closing the modal, and moving from one template to the next should reflect any filtering we have done. 

## Summary


Explore Templates now displays a modal when clicking "See Details" on any template. Keyboard navigable. Moving to the next template reflects any filtering. We removed the Related Templates grid since it was below the fold and seemed a bit out of date at this point. 

## Relevant Technical Choices


`tempateActions` now contains `handleDetailsToggle` to open and close modal, as well as using the templates (`createStoryFromTemplate`) and switching between templates while in modal. Moved templateDetails in modal folder. Added modal karma tests, and updated related tests. Removed routing since the details page no longer exists as its own page. 

## To-do


- [x] update stories
- [x] ui design updates -> overall padding, white space, disabled button, and close button.
- [x]  fix   ● Template Details <Header /> › should trigger mockHandleCtaClick when cta is clicked test
- [x] update pr description
- [x] make additional ticket for url params handling -> ticket [#9996](https://github.com/google/web-stories-wp/issues/9996)

## User-facing changes


![Kapture 2021-12-14 at 11 49 52](https://user-images.githubusercontent.com/1820266/146061259-d4d61dd9-c2ba-43cc-a258-981cc22cde29.gif)


## Testing Instructions






This PR can be tested by following these steps:

1. Go to Explore templates, click on `See Details` button
2. Modal should pop up. Arrows will show the next/prev template
3. Clicking on a thumbnail should update the preview image to that page
4. `Use template` button should create a new story with the current template
5.  Should be keyboard navigable. `esc` should close modal, `tab` will take you through interactive elements. While on image thumbnails, arrows should move you through each thumbnail while enter will select it. 
6. next/prev arrows are disable when at the beginning or ending of the template list. 
7. When modal is close, if you narrow down the template list by searching, the modal should reflect only templates that are remaining. 


## Reviews

### Does this PR have a security-related impact?
no


### Does this PR change what data or activity we track or use?
Yes?


### Does this PR have a legal-related impact?
no


## Checklist




---



Fixes #9671
